ir/instr.h

```diff
@@ -173,17 +173,19 @@ class FnCall final : public Instr {
 public:
   enum Flags { None = 0, NoRead = 1 << 0, NoWrite = 1 << 1, ArgMemOnly = 1 << 2,
                NNaN = 1 << 3 };
+  enum ArgFlags { ArgNone = 0, ArgByVal = 1 << 0 };
 private:
   std::string fnName;
   std::vector<Value*> args;
```

Comment:
I think it's preferable to make this a vector<pair<>> instead. Simplifies a bunch of code & ensure the invariant that every arg has an attribute.